hypothesis-python/tests/conjecture/common.py

```diff
@@ -67,3 +71,142 @@ def accept(f):
+        # should consider this 1024 limit when drawing min_value and
+        # max_value.
+        # This will allow us to remove too_slow and filter_too_much suppressions
+        # at usage sites.
+        assume(width <= 1024)
+
+        weights = draw(
+            st.lists(
+                # weights doesn't play well with super small floats.
+                st.floats(
+                    min_value=0.1, max_value=1, allow_nan=False, allow_infinity=False
+                ),
+                min_size=width,
+                max_size=width,
+            )
+        )
+
+    if forced is not None:
+        assume((forced - shrink_towards).bit_length() < 128)
+
+    return {
+        "min_value": min_value,
+        "max_value": max_value,
+        "shrink_towards": shrink_towards,
+        "weights": weights,
+        "forced": forced,
+    }
+
+
+@st.composite
+def draw_string_kwargs(draw, *, use_min_size=True, use_max_size=True, use_forced=False):
+    intervals = draw(Intervals)
+    alphabet = [chr(c) for c in intervals]
+    forced = draw(st.text(alphabet)) if use_forced else None
+    min_size = 0
+    max_size = None
+
+    if use_min_size:
+        # cap to some reasonable min size to avoid overruns.
+        n = 100
+        if forced is not None:
+            n = min(n, len(forced))
+
+        min_size = draw(st.integers(0, n))
+
+    if use_max_size:
+        n = min_size if forced is None else max(min_size, len(forced))
+        max_size = draw(st.integers(min_value=n))
+        # cap to some reasonable max size to avoid overruns.
+        max_size = min(max_size, min_size + 100)
+
+    return {
+        "intervals": intervals,
+        "min_size": min_size,
+        "max_size": max_size,
+        "forced": forced,
+    }
+
+
+@st.composite
+def draw_bytes_kwargs(draw, *, use_forced=False):
+    forced = draw(st.binary()) if use_forced else None
+    # be reasonable with the number of bytes we ask for. We only have BUFFER_SIZE
+    # to work with before we overrun.
+    size = (
+        draw(st.integers(min_value=0, max_value=100)) if forced is None else len(forced)
+    )
+
+    return {"size": size, "forced": forced}
+
+
+@st.composite
+def draw_float_kwargs(draw, *, use_forced=False):
+    forced = draw(st.floats()) if use_forced else None
+
+    max_n = forced if not math.isnan(forced) else None
+    min_value = draw(st.floats(max_value=max_n, allow_nan=False))
+
+    min_n = min_value if forced is None else max(min_value, forced)
+    max_value = draw(st.floats(min_value=min_n, allow_nan=False))
+
+    return {"min_value": min_value, "max_value": max_value, "forced": forced}
+
+
+@st.composite
+def draw_boolean_kwargs(draw, *, use_forced=False):
+    forced = draw(st.booleans()) if use_forced else None
+    p = draw(st.floats(0, 1, allow_nan=False, allow_infinity=False))
+
+    # avoid invalid forced combinations
+    assume(not (p == 0 and forced is True))
+    assume(not (p == 1 and forced is False))
```

Comment:
```suggestion
    assume(p > 0 or forced is not True)
    assume(p < 1 or forced is not False)
```
I find positive conditions a bit easier to read.  I also tried expressing this as a filter, but that was worse and the whole composite strategy can be retried anyway.